src/tracks/playabletrack/wavetrack/ui/WaveTrackShifter.cpp

```diff
@@ -21,8 +21,22 @@ class WaveTrackShifter final : public TrackShifter {
    HitTestResult HitTest(
       double time, const ViewInfo &viewInfo, HitTestParams* ) override
    {
-      auto pClip = mpTrack->GetClipAtTime( time );
-
+      auto pClip = [&]() {
+         auto ts = mpTrack->TimeToLongSamples(time);
+         
+         for (auto clip : mpTrack->GetClips())
+         {
+            const auto c0 = mpTrack->TimeToLongSamples(clip->GetStartTime());
+            //count clip length until the beginning of the last sample, to exclude
+            //overlapping (see TimeToLongSamples implementation) with the first sample
+            //of the next clip
+            const auto c1 = mpTrack->TimeToLongSamples(clip->GetEndTime() - 1.0 / mpTrack->GetRate());
+            if (ts >= c0 && ts <= c1)
```

Comment:
... or more simply, just use < not <= and don't adjust c1

Reply:
Yep, thanks! It's a leftover I overlooked from the previous iteration where I used time borders instead of samples